src/ServiceControl/Monitoring/Web/EndpointsSettingsController.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace ServiceControl.Monitoring;
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Persistence;
+using ServiceBus.Management.Infrastructure.Settings;
+
+public class EndpointSettingsUpdateModel
+{
+    public bool TrackInstances { get; set; }
+}
+
+public class SettingsData
+{
+    public string Name { get; set; }
+    public bool TrackInstances { get; set; }
+}
+
+[ApiController]
+[Route("api")]
+public class EndpointsSettingsController(
+    IEndpointSettingsStore dataStore, Settings settings)
+    : ControllerBase
+{
+    [Route("endpointssettings")]
+    [HttpGet]
+    public async IAsyncEnumerable<SettingsData> Endpoints([EnumeratorCancellation] CancellationToken token)
+    {
+        await using IAsyncEnumerator<EndpointSettings> enumerator =
+            dataStore.GetAllEndpointSettings().GetAsyncEnumerator(token);
+        bool noResults = true;
+        while (await enumerator.MoveNextAsync())
+        {
+            noResults = false;
+            yield return new SettingsData
+            {
+#pragma warning disable IDE0055
```

Comment:
what does this look like if you don't disable IDE0055?

Reply:
This is my IDE playing up and reformatting this in a weird way.
Again the .editorconfig is flaw